Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java

```diff
@@ -400,60 +332,193 @@ else if (world.equals(craft.getWorld())
-            Location midpoint = new Location(
-                    craft.getWorld(),
-                    (oldHitBox.getMaxX() + oldHitBox.getMinX()) / 2.0,
-                    (oldHitBox.getMaxY() + oldHitBox.getMinY()) / 2.0,
-                    (oldHitBox.getMaxZ() + oldHitBox.getMinZ()) / 2.0);
-            for (Entity entity : craft.getWorld().getNearbyEntities(midpoint,
-                    oldHitBox.getXLength() / 2.0 + 1,
-                    oldHitBox.getYLength() / 2.0 + 2,
-                    oldHitBox.getZLength() / 2.0 + 1
-            )) {
-
-                if (entity instanceof HumanEntity) {
-                    InventoryView inventoryView = ((HumanEntity) entity).getOpenInventory();
-                    if (inventoryView.getType() != InventoryType.CRAFTING) {
-                        Location l = Movecraft.getInstance().getWorldHandler().getAccessLocation(inventoryView);
-                        if (l != null) {
-                            MovecraftLocation location = new MovecraftLocation(l.getBlockX(), l.getBlockY(), l.getBlockZ());
-                            if (oldHitBox.contains(location)) {
-                                location = location.translate(dx, dy, dz);
-                                updates.add(new AccessLocationUpdateCommand(inventoryView, location.toBukkit(world)));
-                            }
-                        }
-                    }
-                }
-
-                if ((entity.getType() == EntityType.PLAYER && !(craft instanceof SinkingCraft))) {
-                    CraftTeleportEntityEvent e = new CraftTeleportEntityEvent(craft, entity);
-                    Bukkit.getServer().getPluginManager().callEvent(e);
-                    if (e.isCancelled())
-                        continue;
-
-                    EntityUpdateCommand eUp = new EntityUpdateCommand(entity, dx, dy, dz, 0, 0,
-                            world, sound, volume);
-                    updates.add(eUp);
-                } else if (!craft.getType().getBoolProperty(CraftType.ONLY_MOVE_PLAYERS)
-                        || entity.getType() == EntityType.PRIMED_TNT) {
-                    CraftTeleportEntityEvent e = new CraftTeleportEntityEvent(craft, entity);
-                    Bukkit.getServer().getPluginManager().callEvent(e);
-                    if (e.isCancelled())
-                        continue;
+        preventsTorpedoRocketsPilots();
+        captureYield(harvestedBlocks);
+    }
 
-                    EntityUpdateCommand eUp = new EntityUpdateCommand(entity, dx, dy, dz, 0, 0, world);
-                    updates.add(eUp);
-                }
-            }
-        } else {
+    private void preventsTorpedoRocketsPilots() {
+        if (craft instanceof SinkingCraft
+                && craft.getType().getBoolProperty(CraftType.ONLY_MOVE_PLAYERS)
+                || !craft.getType().getBoolProperty(CraftType.MOVE_ENTITIES)) {
```

Comment:
Defining the following symbols:
- `a`: `craft instanceof SinkingCraft`
- `b`: `craft.getType().getBoolProperty(CraftType.ONLY_MOVE_PLAYERS)`
- `c`: `craft.getType().getBoolProperty(CraftType.MOVE_ENTITIES)`

Inverting the original `!(a && b) && c` means this should be `(a && b) || !c`, or a better way to write it `!c || (a && b)`.

Reply:
fell twice same thing... fixing it